bazel/thirdparty/krb5.BUILD

```diff
@@ -22,8 +22,10 @@ configure_make(
     ],
     lib_source = ":srcs",
     out_shared_libs = [
-        "libkrb5.so",
-        "libgssapi_krb5.so",
+        "libgssapi_krb5.so.2",
+        "libk5crypto.so.3",
+        "libkrb5.so.3",
+        "libkrb5support.so.0",
```

Comment:
it's odd that we need the versioned forms of these, no?

Reply:
I need to learn more about linking in general to say so for sure. But ldd at least reported that the unversioned one wasn't being picked up. I can investigate more on Monday

Reply:
it probably has something to do with the way autoconf is working. thinking about this, i wonder if we deleted the versioned forms if the dependee would pick these up. anyway, nbd maybe some day we can rewrite krb5 in native bazel.

Reply:
Yeah life is easier with a pure bazel build. I suspect it has something to do with the unversioned being a symlink to the versioned, but can look more later.

Reply:
BTW it looks like our existing build also uses the versioned .so

```
$ ls unpacked_rp/lib/*
unpacked_rp/lib/ld.so                               unpacked_rp/lib/libboost_random.so.1.82.0  unpacked_rp/lib/libgmp.so.10         unpacked_rp/lib/libm.so.6
unpacked_rp/lib/libatomic.so.1                      unpacked_rp/lib/libboost_thread.so.1.82.0  unpacked_rp/lib/libgnutls.so.30      unpacked_rp/lib/libnettle.so.8
unpacked_rp/lib/libboost_atomic.so.1.82.0           unpacked_rp/lib/libc++.so.1                unpacked_rp/lib/libgssapi_krb5.so.2  unpacked_rp/lib/libnuma.so.1
unpacked_rp/lib/libboost_chrono.so.1.82.0           unpacked_rp/lib/libc++abi.so.1             unpacked_rp/lib/libhogweed.so.6      unpacked_rp/lib/libpciaccess.so.0
unpacked_rp/lib/libboost_container.so.1.82.0        unpacked_rp/lib/libc.so.6                  unpacked_rp/lib/libhwloc.so.15       unpacked_rp/lib/libsctp.so.1
unpacked_rp/lib/libboost_date_time.so.1.82.0        unpacked_rp/lib/libcares.so.2              unpacked_rp/lib/libk5crypto.so.3     unpacked_rp/lib/liburing.so.2
unpacked_rp/lib/libboost_filesystem.so.1.82.0       unpacked_rp/lib/libcom_err.so.3            unpacked_rp/lib/libkrb5.so.3         unpacked_rp/lib/libxml2.so.2
unpacked_rp/lib/libboost_iostreams.so.1.82.0        unpacked_rp/lib/libcryptopp.so.8.7         unpacked_rp/lib/libkrb5support.so.0  unpacked_rp/lib/libz.so.1
unpacked_rp/lib/libboost_program_options.so.1.82.0  unpacked_rp/lib/libgcc_s.so.1              unpacked_rp/lib/liblz4.so.1
```